### PR TITLE
Implement a class for managing UTXO

### PR DESCRIPTION
**Pull request by MichaelKim20**
This calculates the sum of the UTXO amounts.
This also provides UTXOs for the required amount.
This is dependent on PR #67 

Related to #63

### CODE REVIEW

File: src/modules/utils/UTXOManager.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+
+    Contains class for managing UTXO
+
+    This class manages UTXO received from Stoa.
```
Comment (TrustHenry) [inline src/modules/utils/UTXOManager.ts:5]: This class may not be related to `Stoa`.
It's seen as a UTXO management tool.

File: src/modules/utils/UTXOManager.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    {
+        this.items = [];
+        this.add(data);
+    }
+
+    /**
+     * Add UTXOs. If it already exists, ignore it.
+     * @param data The array of `UnspentTxOutput`
+     */
+    public add (data: Array<UnspentTxOutput>)
```
Comment (TrustHenry) [inline src/modules/utils/UTXOManager.ts:48]: There are no additional tests for `add` other than tests in the UTXOManager constructor.

File: src/modules/utils/UTXOManager.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    {
+        let old_length = this.items.length;
+        this.items.push(
+            ...data
+                .filter(n => this.items.find(m => (m.utxo.data.compare(n.utxo.data) === 0)) === undefined)
```
Comment (TrustHenry) [inline src/modules/utils/UTXOManager.ts:53]: Is there no possibility of a reference error?

**Review by TrustHenry — COMMENTED** [review]
That's good. a few comments

File: src/modules/utils/UTXOManager.ts (hunk 2 above)
Comment (MichaelKim20) [reply src/modules/utils/UTXOManager.ts:48]: I added it somewhere else, but it's not here. I'll add it.

File: src/modules/utils/UTXOManager.ts (hunk 3 above)
Comment (MichaelKim20) [reply src/modules/utils/UTXOManager.ts:53]: n functions such as filter, find, and so on, all parameters pass instances of the object.

**Comment by MichaelKim20** [comment]
I fixed it all

**Review by TrustHenry — APPROVED** [review]
LGTM

File: src/modules/utils/UTXOManager.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    constructor (data: Array<UnspentTxOutput>)
+    {
+        this.items = [];
+        this.add(data);
+    }
+
+    /**
+     * Add UTXOs. If it already exists, ignore it.
+     * After add, it is sorted in ascending order of unlock_height.
```
Comment (TrustHenry) [inline src/modules/utils/UTXOManager.ts:46]: Second sort condition, Amount description is missing.

Comment (MichaelKim20) [reply src/modules/utils/UTXOManager.ts]: That's a good point. It has just been modified.

**Review by TrustHenry — APPROVED** [review]

**Pull request merged**